core/src/main/java/org/elasticsearch/indices/fielddata/cache/IndicesFieldDataCache.java

```diff
@@ -68,18 +69,17 @@ public IndicesFieldDataCache(Settings settings, IndicesFieldDataCacheListener in
         super(settings);
         this.threadPool = threadPool;
         this.indicesFieldDataCacheListener = indicesFieldDataCacheListener;
-        final String size = settings.get(INDICES_FIELDDATA_CACHE_SIZE_KEY, "-1");
-        final long sizeInBytes = settings.getAsMemory(INDICES_FIELDDATA_CACHE_SIZE_KEY, "-1").bytes();
+        final long sizeInBytes = INDICES_FIELDDATA_CACHE_SIZE_KEY.get(settings).bytes();
         CacheBuilder<Key, Accountable> cacheBuilder = CacheBuilder.<Key, Accountable>builder()
                 .removalListener(this);
         if (sizeInBytes > 0) {
             cacheBuilder.setMaximumWeight(sizeInBytes).weigher(new FieldDataWeigher());
         }
 
-        logger.debug("using size [{}] [{}]", size, new ByteSizeValue(sizeInBytes));
+        logger.debug("using size [{}]", new ByteSizeValue(sizeInBytes));
```

Comment:
Not sure either especially if it's in debug mode. Maybe we don't trust the parsing of ByteSizeValue ;) ? Anyway I removed it.